core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -364,7 +364,24 @@ private[chisel3] class ChiselContext() {
 }
 
 private[chisel3] class DynamicContext(val annotationSeq: AnnotationSeq, val throwOnFirstError: Boolean) {
+  val importedDefinitionAnnos = annotationSeq.collect { case a: ImportedDefinitionAnnotation[_] => a }
+
+  // Ensure there are no repeated names for imported Definitions
+  val importedDefinitionNames = importedDefinitionAnnos.map { a => a.importedDefinition.proto.name }
+  if (importedDefinitionNames.distinct.length < importedDefinitionNames.length) {
+    val duplicates = importedDefinitionNames.diff(importedDefinitionNames.distinct).mkString(", ")
+    throwException(s"Expected distinct imported Definition names but found duplicates for: $duplicates")
+  }
```

Comment:
Not sure what you mean by "conflict". The top-level module should be able to access the submodules uniquely, since the `Definition`s are now forced to have unique names.

The submodules also won't conflict in the `Builder`, since each `Definition` is built in a separate call to `ChiselStage`.

Reply:
you could in theory pass in all the Modules as definitions from the first elaboration, then potential conflicts could be detected/avoided. If yo only pass in the top level definition you'd likely have issues in firrtl/verilog linking later which is still only hypothetical

Reply:
Yes, I think the question is how we would have to do linking. @sequencer and @mwachs5  make good points, where we won't be able to link easily, without renaming conflict names when combining the Verilog. We could avoid this by adding submodule names to the duplicates.

This is probably the best intermediate solution, because it guarantees no name conflicts.